evalml/pipelines/components/transformers/imputers/simple_imputer.py

```diff
@@ -68,26 +64,26 @@ def transform(self, X, y=None):
             y (ww.DataColumn, pd.Series, optional): Ignored.
 
         Returns:
-            pd.DataFrame: Transformed X
+            ww.DataTable: Transformed X
         """
         X = _convert_to_woodwork_structure(X)
         X = _convert_woodwork_types_wrapper(X.to_dataframe())
-        # Convert None to np.nan, since None cannot be properly handled
-        X = X.fillna(value=np.nan)
 
         # Return early since bool dtype doesn't support nans and sklearn errors if all cols are bool
         if (X.dtypes == bool).all():
-            return X
+            return _convert_to_woodwork_structure(X)
+
         X_null_dropped = X.copy()
         X_null_dropped.drop(self._all_null_cols, axis=1, errors='ignore', inplace=True)
-        category_cols = X_null_dropped.select_dtypes(include=['category']).columns
         X_t = self._component_obj.transform(X)
         if X_null_dropped.empty:
-            return pd.DataFrame(X_t, columns=X_null_dropped.columns)
+            X_t = pd.DataFrame(X_t, columns=X_null_dropped.columns)
+            return _convert_to_woodwork_structure(X_t)
+
         X_t = pd.DataFrame(X_t, columns=X_null_dropped.columns)
-        if len(category_cols) > 0:
-            X_t[category_cols] = X_t[category_cols].astype('category')
-        return X_t
+        X_t = X_t.infer_objects()
+        X_t.index = X_null_dropped.index
```

Comment:
Imputer handling indices and restoring to original was moved to SimpleImputer. This way, we support the same behavior for SimpleImputer and don't need to maintain it in two separate places since the Imputer is just a combination of SimpleImputers.